File: frontend-web/E-commerce/src/app/types/product.ts
import {Category} from "./category";
import {SustainabilityCriteria} from "./sustainabilityCriteria";

export interface Product {
  name: string,
  id?: number,
  price: number,
  description: string,
  labels: string[],
  sustainabilityCriteria: SustainabilityCriteria[],
  category: Category
}

File: backend-java/E-commerce/shopping-cart-service/src/main/java/be/pxl/services/domain/Product.java
package be.pxl.services.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@Entity
@Builder
@NoArgsConstructor
@Table(name = "product")
public class Product {

    @Id
    private long id;

    private String name;

    private float price;

    @Enumerated(value = EnumType.STRING)
    private Category category;
}
